actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -671,12 +675,18 @@ def use_relative_controller!
 
         # Remove leading slashes from controllers
         def normalize_controller!
-          @options[:controller] = controller.sub(%r{^/}, ''.freeze) if controller
+          if controller
+            if m = controller.match(/\A\/(?<controller_without_leading_slash>.*)/)
+              @options[:controller] = m[:controller_without_leading_slash]
+            else
+              @options[:controller] = controller
+            end
+          end
         end
 
         # Move 'index' action from options to recall
         def normalize_action!
-          if @options[:action] == 'index'
+          if @options[:action] == 'index'.freeze
```

Comment:
This explains the freeze method for strings 
http://tmm1.net/ruby21-fstrings/

This article explains why you might want I to use frozen strings for performance with some benchmarks http://www.sitepoint.com/unraveling-string-key-performance-ruby-2-2/